src/utils.rs

```diff
@@ -647,8 +647,9 @@ pub(crate) fn trim_left_preserve_layout(
 
 /// Based on the given line, determine if the next line can be indented or not.
 /// This allows to preserve the indentation of multi-line literals.
-pub(crate) fn indent_next_line(kind: FullCodeCharKind, _line: &str, config: &Config) -> bool {
-    !(kind.is_string() || (config.version() == Version::Two && kind.is_commented_string()))
+pub(crate) fn indent_next_line(kind: FullCodeCharKind, line: &str, config: &Config) -> bool {
+    !((kind.is_string() && !(config.format_strings() && line.ends_with('\\')))
+        || (config.version() == Version::Two && kind.is_commented_string()))
```

Comment:
Thanks for submitting this PR!

Presumably by making this change ``indent_next_line`` returns true for the input referenced in the original issue (the input you included as a test case). Just trying to wrap my head around the added conditional check and I was hoping you could give a little more detail regarding the logic of your solution. Maybe a clarifying comment would be useful here?

Reply:
I added a comment to the function explaining the added conditional. Do you think I should split the conditional up, say into an `if/else` statement to potentially make it clearer?

Reply:
Thanks for taking the time to add some additional context to the doc comment. I'm still a little confused by the ``line.ends_with('\\')`` check and why that was needed. I should have been more descriptive about what I was confused by. Also, could you add an additional test case where the string is 2x or 3x longer. I want to see what happens when we need to wrap more than once.

Reply:
I've refactored the function to be an `if/else` which should be a bit more readable when coupled with the added comments.

I've also added a test with lorem ipsum as the string which was long enough to require wrapping a few times.

Reply:
Thanks a lot for the quick turn around on this. I think the current implementation (using ``if/else``), along with the added comment gives plenty of context for what's going on here.

@calebcartwright I'm happy to move forward with these changes, but please let us know if there's anything else you'd like to see from this PR.